### DISCUSSION TITLE
Compatibility with MPP Solar LVX-6048?

### DISCUSSION THREAD

**Discussion by derekyle** [syssi/esphome-pipsolar#20] [Q&A] [answered]
@syssi I was wondering if you have any idea about these new MPP Solar LVX-6048 and the protocol they use? They are similar in a lot of ways and have an rj45 to serial adapter included, but I have not been able to to get anything to work as of yet. Here are some different brandings of what I believe to all be the same, new model:

Voltronic Axpert King II
https://voltronicpower.com/en-US/Product/Detail/Axpert-King-II-5KVA

MPP Solar LVX6048
https://www.mppsolar.com/v3/lvx6048/

SunGoldPower BluePower TP6048 <--I have this one
https://sungoldpower.com/collections/hybrid-solar-inverter/products/6000w-48v-hybrid-solar-inverter-split-phase-120-240vac

One of the manuals: https://cdn.shopifycdn.net/s/files/1/0323/4090/2025/files/BluePower_TP6048.pdf

Any idea on this? I have not been able to find any details on their protocol, but it seems unlikely that it would be different from the earlier Voltronics that are supposedly compatible with this PIP4048 project.

**Comment by SeByDocKy** [comment, marked as answer]
FYI the pipsolar ESPhome integration works fine with the Axpert King II...

**Reply by derekyle** [reply]
Oh, awesome. Do you have any details on wiring up the rj45 to a rs232 to ttl converter? Is it a standard pin-out of some sort? I have tried with the pip documentation in this project, but I can't seem to get it to work. It could be my d9 rs232 to ttl converter, but I want to confirm my pins are correct first.

**Comment by derekyle** [comment]
Just updating this in case others are looking. I believe that the Axpert King II is not actually the same unit as the other units listed here. The SunGoldPower TP6048 (and I believe MPP Solar LVX6048) seem to use the PI18 protocol which is currently available under a different branch: https://github.com/syssi/esphome-pipsolar/tree/pi18 

Additionally, after several misses, this converter worked for me to convert the RS232 signal to TTL. The DB9 pinout matches what you need to connect with the rj45 to db9 connector that comes with the unit (the SunGoldPower TP6048 at least). Note that you will need the male version. https://www.amazon.com/dp/B07BJJ1T5G The same unit seems to be available on aliexpress for a bit cheaper. I'll attach an image of the unit in case the amazon listing disappears.

![image](https://github.com/syssi/esphome-pipsolar/assets/2208099/7bc7701f-f517-4a94-9274-0db9940ce462)
